wpep0002.md

```diff
@@ -0,0 +1,58 @@
+| Key | Value |
+| ---: | :--- |
+| **Title** | Improve distributed capabilities |
+| **Number** | 0002 |
+| **Date** | 2019-10-21 |
+| **Authors** | Frederic Lepied <flepied@wazo.io> |
+| **Status** | proposed |
+
+## Abstract
+
+Redesign the way services find each other and their configurations.
+
+## Motivation
+
+To be able to cope with container deployements like kubernetes, swarm
+or docker-compose as well as distributed deployments on vm or bare
+metal, the startup sequences of the services need to be more dynamic
+without anything being done at install time.
+
+## Proposition
+
+Use Consul to discover services and settings.
+
+While the needed services are not started (nothing in Consul), the
+service shound wait for them to start and publish their informations.
+
+If a service needs a database, it should manage it at startup: create
+it if it doesn't exist and do the modifications if a database schema
+change is needed.
+
+The setup of the services themselves like RabbitMQ in cluster mode,
+load balancing database is out of scope of this proposition. What is
+exposed by this proposition is only the URI of the service not the way
+it has been deployed.
+
+To summarize, nothing should be done by the packages or the Ansible
+installer at install time that is deployment specific. For example, at
+install time, we must not generate SSL certs, create or manipulate
+databases or set any deployment specific parameter.
+
+For non Consul aware services like Asterisk, Kamailio, RTPEngine,
+PostgreSQL or RabbitMQ, we have 2 options. First one is to create a
+wrapper which is interfacing with Consul to publish the state and get
+the needed parameters. The second option is to write a plugin for the
+projects that support it like Asterisk or Kamailio to interface with
+Consul.
+
+### Details
+
+All the services, plugins and wrappers that are Consul aware must use
+the `CONSUL_HTTP_ADDR` environment variable to find a Consul server to
```

Comment:
Since we do not put an address in this variable, can we name it to something more generic like `CONSUL_HOST`.
Also, do we need to be able to set:
* port
* scheme
* verify certificate (true/false/custom certificate)